OPHD/States/MapViewStateTurn.cpp

```diff
@@ -165,36 +165,57 @@ void MapViewState::updateMorale()
 
 	// POSITIVE MORALE EFFECTS
 	// =========================================
-	mCurrentMorale += mPopulation.birthCount();
-	mCurrentMorale += structureManager.getCountInState(Structure::StructureClass::Park, StructureState::Operational);
-	mCurrentMorale += structureManager.getCountInState(Structure::StructureClass::RecreationCenter, StructureState::Operational);
-
-	int food_production = structureManager.getCountInState(Structure::StructureClass::FoodProduction, StructureState::Operational);
-	mCurrentMorale += food_production > 0 ? food_production : -5;
-
-	mCurrentMorale += structureManager.getCountInState(Structure::StructureClass::Commercial, StructureState::Operational);
+	const int birthCount = mPopulation.birthCount();
+	const int parkCount = structureManager.getCountInState(Structure::StructureClass::Park, StructureState::Operational);
+	const int recreationCount = structureManager.getCountInState(Structure::StructureClass::RecreationCenter, StructureState::Operational);
+	const int foodProducingStructures = structureManager.getCountInState(Structure::StructureClass::FoodProduction, StructureState::Operational);
+	const int commercialCount = structureManager.getCountInState(Structure::StructureClass::Commercial, StructureState::Operational);
+	
 
 	// NEGATIVE MORALE EFFECTS
 	// =========================================
-	mCurrentMorale -= mPopulation.deathCount();
-	mCurrentMorale -= structureManager.disabled();
-	mCurrentMorale -= structureManager.destroyed();
-
-	int residentialMoraleHit = mPopulationPanel.capacity() / 100;
-
-	// Ensure that there is always a morale hit if residential capacity is more than 100%.
-	if (mPopulationPanel.capacity() > 100 && residentialMoraleHit < constants::MINIMUM_RESIDENCE_OVERCAPACITY_HIT) { residentialMoraleHit = constants::MINIMUM_RESIDENCE_OVERCAPACITY_HIT; }
+	const int deathCount = mPopulation.deathCount();
+	const int structuresDisabled = structureManager.disabled();
+	const int structuresDestroyed = structureManager.destroyed();
+	
+	const int residentialOverCapacityHit = mPopulation.size() > mResidentialCapacity ? -2 : 0;
 
 	auto& residences = NAS2D::Utility<StructureManager>::get().structureList(Structure::StructureClass::Residence);
+	int bioWasteAccumulation = 0;
 	for (auto residence : residences)
 	{
 		Residence* unit = static_cast<Residence*>(residence);
-		if (unit->wasteOverflow() > 0) { mCurrentMorale -= 2; } /// \fixme magic number
+		if (unit->wasteOverflow() > 0) { ++bioWasteAccumulation; }
 	}
 
-	mCurrentMorale -= residentialMoraleHit;
+	// positive
+	mCurrentMorale += birthCount;
+	mCurrentMorale += parkCount;
+	mCurrentMorale += recreationCount;
+	mCurrentMorale += commercialCount;
+	mCurrentMorale += foodProducingStructures > 0 ? 0 : -5;
```

Comment:
The `foodProducingStructures` line is actually a negative. Might be more clearly a negative if it was written as:
```cpp
mCurrentMorale -= foodProducingStructures > 0 ? 0 : 5;
```

Btw, any thoughts on if it's more clear or not to include parenthesis around ternary conditions?
```cpp
mCurrentMorale -= (foodProducingStructures > 0) ? 0 : 5;
```

Reply:
I would agree that parenthesis around ternary operators is generally more clear and is usually the way I try to write them. I would consider this a stylistic choice that should be encouraged.

And yes, it would make sense to write this in the negative. I think I was considering having a positive effect on growing food stores but that was unrelated to the number of actual food production structures versus actually seeing a positive trend, it just didn't translate over.

I think I'm going to open a separate change set / pr for code audits like these to clean up the things you notice that could be done better.